tests/api/controllers/cache_checker/cache_checker.c

```diff
@@ -0,0 +1,117 @@
+#include <stdio.h>
+#include <webots/camera.h>
+#include <webots/distance_sensor.h>
+#include <webots/robot.h>
+#include <webots/supervisor.h>
+
+#include "../../../lib/ts_assertion.h"
+#include "../../../lib/ts_utils.h"
+
+#define TIME_STEP 64
+#define NB_TEXTURES 4
+#define NB_MESHES 3
+
+static WbDeviceTag cameras[NB_TEXTURES];
+static WbDeviceTag distance_sensors[NB_MESHES];
+
+void test_camera_color(int test_number, const int expected_color[3]) {
+  int r, g, b;
+  const int width = wb_camera_get_width(cameras[0]);
+
+  for (int i = 0; i < NB_TEXTURES; ++i) {
+    const unsigned char *image = wb_camera_get_image(cameras[i]);
+    r = wb_camera_image_get_red(image, width, 32, 32);
+    g = wb_camera_image_get_green(image, width, 32, 32);
+    b = wb_camera_image_get_blue(image, width, 32, 32);
+    ts_assert_color_in_delta(
+      r, g, b, expected_color[0], expected_color[1], expected_color[2], 2,
+      "In test %d: wrong texture color in object %d, received color = (%d, %d, %d) but expected color = (%d, %d, %d)",
+      test_number, i, r, g, b, expected_color[0], expected_color[1], expected_color[2]);
+  }
+}
+
+void test_distance(int test_number, const double expected_distance) {
+  for (int i = 0; i < NB_MESHES; ++i) {
+    const double distance = wb_distance_sensor_get_value(distance_sensors[i]);
+    ts_assert_double_in_delta(distance, expected_distance, 1e-5,
+                              "In test %d: wrong mesh is applied (distance is %f instead of %f)", distance, expected_distance);
```

Comment:
```suggestion
                              "In test %d: wrong mesh %d is applied (distance is %f instead of %f)", test_number, i, distance, expected_distance);
```